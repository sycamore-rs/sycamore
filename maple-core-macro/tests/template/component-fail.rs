#![allow(non_snake_case)]

use maple_core::prelude::*;

fn Component<G: GenericNode>() -> TemplateResult<G> {
    template! {
        div
    }
}

fn compile_fail<G: GenericNode>() {
    let _: TemplateResult<G> = template! { UnknownComponent() };

    let _: TemplateResult<G> = template! { Component };
    let _: TemplateResult<G> = template! { Component(1) };
}

fn main() {}
