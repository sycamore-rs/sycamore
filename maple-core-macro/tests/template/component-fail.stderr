error: unexpected end of input, expected parentheses
  --> $DIR/component-fail.rs:14:32
   |
14 |     let _: TemplateResult<G> = template! { Component };
   |                                ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find function, tuple struct or tuple variant `UnknownComponent` in this scope
  --> $DIR/component-fail.rs:12:44
   |
12 |     let _: TemplateResult<G> = template! { UnknownComponent() };
   |                                            ^^^^^^^^^^^^^^^^ not found in this scope

error[E0061]: this function takes 0 arguments but 1 argument was supplied
  --> $DIR/component-fail.rs:15:44
   |
15 |     let _: TemplateResult<G> = template! { Component(1) };
   |                                            ^^^^^^^^^ - supplied 1 argument
   |                                            |
   |                                            expected 0 arguments
   |
note: function defined here
  --> $DIR/component-fail.rs:5:4
   |
5  | fn Component<G: GenericNode>() -> TemplateResult<G> {
   |    ^^^^^^^^^
